### PR TITLE
Pubsub interop with go-libp2p-daemon

### PR DESCRIPTION
**Pull request by mhchia**
Depends on #270 

- Use libp2p daemon bindings to control the daemon, which is slightly modified to support plaintext transport.
- Add utility functions which makes pubsub interop test easier.
- Change to `asyncio.subprocess.Process` from `pexpect` in bindings tests. https://github.com/libp2p/py-libp2p/issues/272
- Subscription and publish are interoperable with Go now.
- Fix https://github.com/libp2p/py-libp2p/issues/191.
- Fix https://github.com/libp2p/py-libp2p/issues/233.
- Fix https://github.com/libp2p/py-libp2p/issues/237.

#### TODOs
- [x] Add pubsub tests.
- [x] Make bindings use the data structures from libp2p, for easier development.
- [x] Add p2pd installation process in CI.
    - [x] Need to use my modified go-libp2p-daemon to support plaintext.
- [x] Kick off printing logs in Daemon.
- [x] Add fixture for Daemon.
- [x] Fix `tests/pubsub/test_gossipsub.py`.
- [x] Clean up debug messages.
- [x] Fix `tests/pubsub/test_pubsub.py`

### CODE REVIEW

**Comment by mhchia** [comment]
Use 3.7-dev in travis CI, for python 3.7.4, instead of 3.7 which is 3.7.1. This is particularly used to solve the issue https://github.com/pypa/pip/issues/5966, due to the use of PEP 508 urls in the dependencies, encountered in the [CI](https://travis-ci.com/libp2p/py-libp2p/jobs/230732772).

File: tests/interop/test_bindings.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    # Test: `disconnect` from Py
+    await host.disconnect(p2pd.peer_id)
+    assert len(await p2pd.control.list_peers()) == 0
+    # Test: connect from Go
+    await connect(p2pd, host)
+    assert len(host.get_network().connections) == 1
+    # Test: `disconnect` from Go
+    await p2pd.control.disconnect(host.get_id())
+    # FIXME: Failed to handle disconnect
+    # assert len(host.get_network().connections) == 0
```
Comment (mhchia) [inline tests/interop/test_bindings.py]: Note: We should detect `disconnect` from the other side, and remove the connection in the dict.

File: tests/interop/utils.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        peers = tuple([pinfo.peer_id for pinfo in pinfos])
+        return peer_id in peers
+    else:  # isinstance(node, IHost)
+        return peer_id in node.get_network().connections
+
+
+async def connect(a: TDaemonOrHost, b: TDaemonOrHost) -> None:
+    # Type check
+    err_msg = (
+        f"Type of type(a)={type(a)} or type(b)={type(b)} is wrong."
```
Comment (NIC619) [inline tests/interop/utils.py]:
```suggestion
        f"Type of a={type(a)} or type of b={type(b)} is wrong."
```

File: tests/interop/conftest.py [hunk 3]
```diff
@@ -22,3 +28,46 @@ def call_proc(cmd, args, logfile=None, encoding=None):
...
+
+
+@pytest.fixture
+def is_gossipsub():
+    return True
+
+
+@pytest.fixture
+async def p2pds(num_p2pds, is_host_secure, is_gossipsub, unused_tcp_port_factory):
+    p2pds = await asyncio.gather(
```
[the remark this answers is not in the document]
Comment (mhchia) [reply tests/interop/conftest.py]: Nice catch! Fixed it in the latest commits.

File: libp2p/network/swarm.py [hunk 4]
```diff
@@ -92,55 +93,55 @@ def set_stream_handler(
...
+
+        try:
+            # Get peer info from peer store
+            addrs = self.peerstore.addrs(peer_id)
+        except PeerStoreError:
+            raise SwarmException(f"No known addresses to peer {peer_id}")
 
         if not addrs:
-            raise SwarmException("No known addresses to peer")
+            raise SwarmException(f"No known addresses to peer {peer_id}")
```
[the remark this answers is not in the document]
Comment (mhchia) [reply libp2p/network/swarm.py]: It seems a little bit weird to me to raise `PeerStoreError` when we have an empty list of addresses for a peer ID.  I prefer to leave this in later discussion.

File: tests/interop/daemon.py [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            *self.args,
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+            bufsize=0,
+        )
+        await self.wait_until_ready()
+        await self.start_printing_logs()
+
+    async def close(self) -> None:
+        if self.is_running:
```
[the remark this answers is not in the document]
Comment (mhchia) [reply tests/interop/daemon.py]: Good catch! Fixed it.

**Pull request merged**